social_network/posts.py

```diff
@@ -1,33 +1,48 @@
 from datetime import datetime
 
 
-# class Post(object):
-#     def __init__(self, text, timestamp=None):
-#         pass
-#
-#     def set_user(self, user):
-#         pass
-
-
-# class TextPost(...):  # Inherit properly
-#     def __init__(self, text, timestamp=None):
-#         pass
-#
-#     def __str__(self):
-#         pass
-
-
-# class PicturePost(...):  # Inherit properly
-#     def __init__(self, text, image_url, timestamp=None):
-#         pass
-#
-#     def __str__(self):
-#         pass
-
-
-# class CheckInPost(...):  # Inherit properly
-#     def __init__(self, text, latitude, longitude, timestamp=None):
-#         pass
-#
-#     def __str__(self):
-#         pass
+class Post(object):
+    user = None
+    
+    def __init__(self, text, timestamp=None):
+        self.text = text
+        self.timestamp = timestamp
+
+    def set_user(self, user):
+        self.user = user
+    
+    def __repr__(self):
+        rep = "{}'s post @{}".format(self.user, self.timestamp.strftime('%A, %b %d, %Y'))
+        return rep
+        
+class TextPost(Post):  # Inherit properly
```

Comment:
I know it's not really that much extra code, but it seemed like a good time to practice using super().

Reply:
I didn't really mark the code I reviewed properly, but I hope you understand what I was trying to say.